cpp/modmesh/buffer/SimpleArray.hpp

```diff
@@ -623,6 +608,37 @@ class SimpleArray
     value_type * body() { return m_body; }
 
 private:
+    void check_c_contiguous(small_vector<size_t> const & shape,
```

Comment:
I like the helper functions.

They do not need to be in the class template because they do not use the template argument.  But it's OK to keep them here for now.

There are other member functions that can be moved outside the class template.  We only need to refactor when we run out of interesting things to do.